core/phases.py

```diff
@@ -25,35 +25,62 @@ def _ready_to_end(self) -> bool:
     def end(self) -> PhaseState:
         if self._ready_to_end():
             self._state = PhaseState.ENDED
-
         return self._state
 
 
+class DealPhase(Phase):
+    def __init__(self, game):
+        super().__init__(game)
+
+    def deal_cards(self) -> bool:
+        if self.__has_enough_cards_to_deal_to_each_player():
+            for player in self._game.players:
+                number_of_cards = player.receives_how_many_cards_at_round_start
+                cards_to_give = self._game.draw_pile[-number_of_cards:]
+                player.add_to_hand_cards(cards_to_give)
+                self._game.draw_pile = self._game.draw_pile[:-number_of_cards]
+            return True
+        else:
+            return False
+
+    @property
+    def __number_of_cards_in_draw_pile(self) -> int:
+        return len(self._game.draw_pile)
```

Comment:
This should reference: number_of_cards_in_draw_pile() on self.__game, right?

Reply:
Yes.